mllib/src/main/scala/org/apache/spark/mllib/stat/correlation/SpearmanCorrelation.scala

```diff
@@ -65,8 +65,8 @@ private[stat] object SpearmanCorrelation extends Correlation with Logging {
         output
       }
       iter.flatMap { case (((j, v), uid), rank) =>
-        // If we see a new value or cachedUids is too big, we flush ids with their average rank.
-        if (j != preCol || v != preVal || cachedUids.size >= 10000000) {
```

Comment:
I think this limitation is on purpose.

I think we could either
1, document this limitation;
2, or add a new parameter for it.

also cc @srowen @WeichenXu123

Reply:
What happens if we don't flush it when `cachedUids.size >= 10000000` ? OOM occurs ?

Reply:
cachedUids.size <= 10000000 can't prevent oom, sortbykey will pull all ties values to a single partitions

Reply:
cachedUids.size <= 10000000 can't prevent oom, sortbykey will pull all ties values to a single partitions

Reply:
with the flush of `cachedUids.size >= N`, does it generate approximate result ? If yes I suggest we can provide a config for cachedUids.size

Reply:
flush of cachedUids.size >= N can't generate approximate results (far from the correct results).  It might seem better to generate a warning without flush

Reply:
> flush of cachedUids.size >= N can't generate approximate results. 

emm, then the logic does not make sense..

Reply:
> I think this limitation is on purpose.
> 
> I think we could either 1, document this limitation; 2, or add a new parameter for it.
> 
> also cc @srowen @WeichenXu123

the intent may be to prevent OOM, but it causes some other problems

- flush when ties size > N can't give a approximate results, and in many cases, is far from the correct results, as it results in distinct average ranks for tied values
- ties size > 10_000_000 are not uncommon. real-world data tends to exhibit skewness with a substantial number of zero values
- alternative workaound may need multiple traversal iterations

Reply:
got it. I agree to remove the size > 10_000_000 condition here.